packages/world/src/systems/MineSystem.sol

```diff
@@ -108,10 +108,14 @@ library MineLib {
   }
 
   function _processMassReduction(EntityId playerEntityId, EntityId minedEntityId) public returns (uint128) {
-    (uint128 toolMassReduction, ) = useEquipped(playerEntityId);
-    uint128 totalMassReduction = energyToMass(PLAYER_MINE_ENERGY_COST) + toolMassReduction;
     uint128 massLeft = Mass._getMass(minedEntityId);
-    return massLeft <= totalMassReduction ? 0 : massLeft - totalMassReduction;
+    uint128 baseMassReduction = energyToMass(PLAYER_MINE_ENERGY_COST);
+    if (massLeft <= baseMassReduction) {
```

Comment:
just trying to understand this, if massLeft is <= baseMassReduction then `useEquipped` is not called, so the tool would not be removed?

Reply:
yee, if the player can mine the block's remaining mass with just the energy they are spending, then we dont use the tool

Reply:
wait but shouldn't we also change the amount of energy transferred to the pool then? lmk if I understand this correctly:
- mine energy cost is removed from the player
- mass is removed from the tool
- mass is removed from the object being mined

shouldn't the amount of energy added to the pool be the sum of those 3 quantities?

Reply:
lets go over this in tuning, added a todo